modules/javafx.controls/src/main/java/javafx/scene/control/skin/LabeledSkinBase.java

```diff
@@ -354,9 +356,10 @@ protected void updateChildren() {
         final double gap = labeled.getGraphicTextGap();
 
         width -= leftInset + rightInset;
-
+        double padding = topInset + bottomInset;
         if (!isIgnoreText()) {
             width -= leftLabelPadding() + rightLabelPadding();
+            padding += topLabelPadding() + bottomLabelPadding();
         }
 
         String cleanText = getCleanText();
```

Comment:
cleanText code block is unnecessary  when isIgnoreText() == true

Reply:
Since `clenText` is used in `textHeight` calculation, this cannot be avoided before the `if-else` block as `if-else` block is not re-ordered now. Please refer to the previous comment for the details on `computePrefHeight()` method.

Reply:
Changed local variable name.
Please refer to my comment at the end regarding `computePrefHeight()` method optimization.